proto/kvrpcpb.proto

```diff
@@ -1034,3 +1038,22 @@ message RawGetKeyTTLResponse {
     uint64 ttl = 3;
     bool not_found = 4;
 }
+
+message RawCASRequest {
+    Context context = 1;
+    bytes key = 2;
+    bytes value = 3;
+    // If this field does not exist, it means that the key-value should only put into database if and only if the key
+    // does not exist.
+    bytes previous_value = 4;
```

Comment:
Actually you can't tell non-exist value from empty value in protobuf.

Reply:
If the value is a empty string, I think it present non-exist

Reply:
Please don't do that. It'll be a legacy problem like this one:

https://github.com/tikv/tikv/issues/6711

Reply:
I suggest adding a new bool field `should_not_exist` to indicate that property.